fakestorage/object.go

```diff
@@ -562,6 +562,8 @@ func (s *Server) downloadObject(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Length", strconv.Itoa(len(content)))
 	w.Header().Set(contentTypeHeader, obj.ContentType)
 	w.Header().Set("X-Goog-Generation", strconv.FormatInt(obj.Generation, 10))
+	w.Header().Set("X-Goog-Hash", fmt.Sprintf("crc32c=%s", obj.Crc32c))
+	// w.Header().Set("X-Goog-Hash", fmt.Sprintf("md5=%s", obj.Md5Hash))
```

Comment:
Do you have any links to the docs that say how google decides to which one to use? Or can it be either?

Reply:
I haven't got anything definite, but I found this, which is mostly about sending the headers in upload requests but does say that "Cloud Storage stores MD5 hashes for all non-composite objects. CRC32Cs are available for all objects."

https://cloud.google.com/storage/docs/xml-api/reference-headers#xgooghash

Reply:
That looks like what the Node client is expecting, yes:

https://github.com/googleapis/nodejs-storage/blob/390469a2283fd85ee3e1121d77505e77e16dbf10/src/file.ts#L1322-L1338

Reply:
Nice, can you update the code to do both?
```suggestion
	w.Header().Set("X-Goog-Hash", fmt.Sprintf("crc32c=%s, md5=%s", obj.Crc32c, obj.Md5Hash))
```

Reply:
I think the separator needs to be just a comma with no space, but yes apart from that. Does it need to handle cases where `obj.Crc32c` or `obj.Md5Hash` aren't set?

Reply:
In the Node client it's possible to set `validation: 'crc32'` or `validation: 'md5'` as an option for the download, which might be a reason to include both in case one has been specified. I'm not sure how common that is, though.